### PR TITLE
apps/utils/date: add messages to show limitation of year, month and week

### PR DESCRIPTION
**Pull request by sunghan-chang**
The date command allows year within [1970,2106] and month, the day
of week allows 3 abbreviation characters.
It it difficult to know these limitations without messages.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [268650e2f610696d04b30b6962086330e7a9bc4e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [268650e2f610696d04b30b6962086330e7a9bc4e] - Code Rule Check (C++) OK.

File: apps/system/utils/kdbg_date.c [hunk 1]
```diff
@@ -325,30 +325,29 @@ int kdbg_date(int argc, char **args)
 #endif
 		ret = date_settime(argc, args);
 	} else {
-#if !(defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED))
```
Comment (btheosam) [inline apps/system/utils/kdbg_date.c]: Why is this removed from here and moved to line 329?

Comment (sunghan-chang) [reply apps/system/utils/kdbg_date.c]: Because 5 lines from start are same between ```#if``` and ```#else```.
Those conditional affects only FORMAT. We don't need to keep same 5 lines at each case.
